Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs

```diff
@@ -32,189 +33,214 @@ public DefaultPointerMediator(IPointerPreferences pointerPrefs)
-                                // It is okay for example to have two near pointers active on a single controller
-                                // like a poke pointer and a grab pointer
-                                continue;
+                                if (!unassignedPointers.Contains(otherPointer))
+                                {
+                                    continue;
+                                }
+
+                                if (otherPointer is IMixedRealityNearPointer)
+                                {
+                                    // Only disable far interaction pointers
+                                    // It is okay for example to have two near pointers active on a single controller
+                                    // like a poke pointer and a grab pointer
+                                    continue;
+                                }
+
+                                otherPointer.IsActive = false;
+                                unassignedPointers.Remove(otherPointer);
                             }
-
-                            otherPointer.IsActive = false;
-                            unassignedPointers.Remove(otherPointer);
                         }
                     }
                 }
-            }
 
-            // All other pointers that have not been assigned this frame
-            // have no reason to be disabled, so make sure they are active
-            foreach (IMixedRealityPointer unassignedPointer in unassignedPointers)
-            {
-                unassignedPointer.IsActive = true;
+                // All other pointers that have not been assigned this frame
+                // have no reason to be disabled, so make sure they are active
+                foreach (IMixedRealityPointer unassignedPointer in unassignedPointers)
+                {
+                    unassignedPointer.IsActive = true;
+                }
             }
         }
 
+        private static readonly ProfilerMarker ApplyCustomPointerBehaviorsPerfMarker = new ProfilerMarker("[MRTK] DefaultPointerMediator.ApplyCustomPointerBehaviors");
+
         private void ApplyCustomPointerBehaviors()
         {
-            if (pointerPreferences != null)
+            using (ApplyCustomPointerBehaviorsPerfMarker.Auto())
             {
-                foreach (IMixedRealityPointer pointer in allPointers)
+                if (pointerPreferences != null)
                 {
-                    ApplyPointerBehavior(pointer, pointerPreferences.GetPointerBehavior(pointer));
+                    foreach (IMixedRealityPointer pointer in allPointers)
+                    {
+                        ApplyPointerBehavior(pointer, pointerPreferences.GetPointerBehavior(pointer));
+                    }
                 }
             }
         }
 
+        private static readonly ProfilerMarker ApplyCustomPointerBehaviorPerfMarker = new ProfilerMarker("[MRTK] DefaultPointerMediator.ApplyCustomPointerBehavior");
```

Comment:
To match the method name:

```suggestion
        private static readonly ProfilerMarker ApplyPointerBehaviorPerfMarker = new ProfilerMarker("[MRTK] DefaultPointerMediator.ApplyPointerBehavior");
```